### ISSUE TITLE
使用 `wsl2` build 失败

### ISSUE THREAD

**Issue by Chanyon** [nature-lang/nature#16]
### build
[nature-lang.org/docs](https://nature-lang.org/docs/prologue/contribution-guide)
cmake -B build-release -DCMAKE_TOOLCHAIN_FILE=$(pwd)/cmake/linux-amd64-toolchain.cmake -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_BUILD_TYPE=Release -DCPACK_OUTPUT_FILE_PREFIX=$(pwd)/release

cmake --build build-release

```shell
/src/binary/elf/linker.c.obj:(.bss+0x58): first defined here
/usr/bin/ld: CMakeFiles/nature.dir/utils/type.c.obj:(.bss+0x60): multiple definition of `ct_symdef_size'; CMakeFiles/nature.dir/src/binary/elf/linker.c.obj:(.bss+0x60): first defined here
/usr/bin/ld: CMakeFiles/nature.dir/utils/type.c.obj:(.bss+0x68): multiple definition of `rt_rtype_ptr'; CMakeFiles/nature.dir/src/binary/elf/linker.c.obj:(.bss+0x68): first defined here
/usr/bin/ld: CMakeFiles/nature.dir/utils/type.c.obj:(.bss+0x70): multiple definition of `rt_fndef_ptr'; CMakeFiles/nature.dir/src/binary/elf/linker.c.obj:(.bss+0x70): first defined here
/usr/bin/ld: CMakeFiles/nature.dir/utils/type.c.obj:(.bss+0x78): multiple definition of `rt_symdef_ptr'; CMakeFiles/nature.dir/src/binary/elf/linker.c.obj:(.bss+0x78): first defined here
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/nature.dir/build.make:721: nature] Error 1
gmake[2]: Leaving directory '/home/projects/nature/build-release'
gmake[1]: *** [CMakeFiles/Makefile2:221: CMakeFiles/nature.dir/all] Error 2
gmake[1]: Leaving directory '/home/projects/nature/build-release'
gmake: *** [Makefile:166: all] Error 2
```

**Comment by weiwenhao** [comment]
![image](https://github.com/nature-lang/nature/assets/24819514/f3d34560-2a3f-4c3b-b0e0-66fa07311bc6)

删掉 utils/custom_links.h 里面的 #include "type.h" 再编译试试。

**Comment by weiwenhao** [comment]
What is the version of gcc, I ran into this issue in gcc11, will be fixed in version 0.4.0-beta

**Comment by Chanyon** [comment]
> What is the version of gcc, I ran into this issue in gcc11, will be fixed in version 0.4.0-beta

gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0
